internal/component_options.go

```diff
@@ -191,9 +191,10 @@ type ChownOptions struct {
 }
 
 type StageDataOptions struct {
-	Name string
-	Id   string
-	Data []byte
+	Name   string
+	Id     string
+	Data   []byte
+	Offset uint64
```

Comment:
Any particular reason to use offset here?

Reply:
Ideally if a component(block_cache) is staging data(write operation) it should also send the offset from where we need to start writing. Next Component of block_cache doesn't necessarily have to be AzStorage.

Reply:
We need offset in encryptor component for our use case because for encryptor it is just file operation to another mount point.